2-api/server.js

```diff
@@ -0,0 +1,227 @@
+const { query, json } = require("express");
+const express = require("express");
+const app = express();
+const { Pool } = require("pg");
+app.use(express.json());
+
+const pool = new Pool({
+  user: "Maziar",
+  host: "localhost",
+  database: "cyf_ecommerce",
+  password: "M@ziar13",
+  port: 5432,
+});
+
+app.get("/", function (req, res) {
+  res.send("Hi there, please type /customers , /suppliers , /products. Have Fun!");
+});
+
+app.get("/customers", function (req, res) {
+  pool
+    .query("SELECT * FROM customers")
+    .then((result) => res.json(result))
+    .catch((e) => console.error(e));
+});
+
+app.get("/suppliers", function (req, res) {
+  pool
+    .query("SELECT * FROM suppliers")
+    .then((result) => res.json(result))
+    .catch((e) => console.error(e));
+});
+
+app.get("/products", function (req, res) {
+  const productName = req.query.product_name;
+  let query =
+    "SELECT products.product_name, product_availability.unit_price, suppliers.supplier_name FROM ((products INNER JOIN product_availability ON products.id = product_availability.prod_id) INNER JOIN suppliers ON suppliers.id = product_availability.supp_id)";
+
+  if (productName) {
+    query = `SELECT * FROM products WHERE product_name LIKE '%${productName}%'`;
+  }
+  pool
+    .query(query)
+    .then((result) => res.json(result.rows))
+    .catch((e) => console.error(e));
+});
+
+app.get("/customers/:customerId", function (req, res) {
+  const customerId = req.params.customerId;
+  const query = "SELECT cust.* FROM customers cust  WHERE cust.id=$1";
+  pool
+    .query(query, [customerId])
+    .then((result) => res.json(result.rows))
+    .catch((e) => console.error(e));
+});
+
+app.post("/customers", function (req, res) {
+  const newCustomerName = req.body.name;
+  const newCustomerAdd = req.body.address;
+  const newCustomerCity = req.body.city;
+  const newCustomerCountry = req.body.country;
+
+  if (!newCustomerName) {
+    return res.status(400).send("Please enter customer name");
```

Comment:
Good error handling - appropriate error message and status code, and you have correctly identified that only the customer name is mandatory in the table.